blog/_posts/2018-03-29-missing.md

```diff
@@ -0,0 +1,394 @@
+[`DataArray`](https://github.com/JuliaStats/DataArrays.jl). In fact, the layout
+of the `DataArray` type is very similar to that of `Array{Union{Missing,T}}`
+described above. The only difference is that it uses a `BitArray` rather than an
+`Array{UInt8}` to indicate whether a value is missing, therefore taking 1 bit per
+entry rather than 8 bits. Even if it consumes more memory, the `Array{UInt8}` mask
+approach is faster (at least in the current state of `BitArray`), and it generalizes
+to `Union`s of more than two types. However, we are aware that other implementations
+such as [PostgreSQL](https://www.postgresql.org/docs/9.5/static/storage-page-layout.html)
+or [Apache Arrow](https://arrow.apache.org/docs/memory_layout.html#null-bitmaps)
+use bitmaps equivalent to `BitArray`.
+
+## Safety and propagation by default
+
+In addition to being generic and efficient, the new missing values support in
+Julia 0.7 aims to provide safety, in the sense that missing values should never
+be silently ignored nor replaced with non-missing values. Missing values are a
+delicate issue in statistical work, and a frequent source of bugs or invalid results.
+Ignoring missing values amounts to performing data imputation, which should never
+happen silently without an explicit request.
+
+Therefore, passing `missing` to a function will always return `missing` or throw
+an error (except for a few special functions presented below). For convenience,
+standard operators and mathematical functions systematically propagate missing
+values:
+
+    julia> 1 + missing
+    missing
+
+    julia> missing^2
+    missing
+
+    julia> cos(missing)
+    missing
+
+    julia> round(missing)
+    missing
+
+    julia> "a" * missing
+    missing
+
+Reduction operations inherit the propagating behavior of basic operators:
+
+    julia> sum([1, missing, 2])
+    missing
+
+    julia> mean([1, missing, 2])
+    missing
+
+On the other hand, indexing into a `Vector` with missing values is an error.
+Missing values are *not* silently skipped, which would be equivalent to assuming
+that they are `false`.
+
+    julia> x = 1:3
+    1:3
+
+    julia> x[[true, missing, false]]
+    ERROR: ArgumentError: unable to check bounds for indices of type Missing
+
+    julia> x[[1, missing]]
+    ERROR: ArgumentError: unable to check bounds for indices of type Missing
+
+Convenience functions are provided to get rid of missing values explicitly.
+First, the `skipmissing` function returns an iterator over the non-missing values
+in the passed collection. It is particularly useful to ignore missing values when
+computing reductions. Call `collect` to obtain a vector with all non-missing values.
+
+    julia> sum(skipmissing([1, missing, 2]))
+    3
+
+    julia> mean(skipmissing([1, missing, 2]))
+    1.5
+
+    julia> collect(skipmissing([1, missing, 2]))
+    2-element Array{Int64,1}:
+    1
+    2
+
+Second, the `coalesce` function returns the first non-missing argument, which
```

Comment:
I know this name is borrowed from somewhere but I'm not sure where; maybe include footnote/pointer for that?

Reply:
Borrowed as in the idea comes from another language? SAS has `coalesce`.

Reply:
Actually it exists in SQL.